scripts/reboot

```diff
@@ -260,4 +276,18 @@ if [ -x ${DEVPATH}/${PLATFORM}/${PLAT_REBOOT} ]; then
 fi
 
 VERBOSE=yes debug "Issuing OS-level reboot ..." >&2
-exec /sbin/reboot $@
+echo "Going to issue reboot command..."
+if [ $# -eq 0 ]; then
+    echo "no argument passed"
+    exec /sbin/reboot $@
+elif [ $# -eq 1 ] && [[ $REBOOT_FORCE == true ]]; then
+    echo "force reboot"
+    exec /sbin/reboot $@
+elif [ $# -eq 2 ] && [[ ${REBOOT_CAUSE} != null ]]; then
+    echo "reboot cause specified"
+    exec /sbin/reboot
+elif [ $# -gt 2 ] && [[ $REBOOT_FORCE == true ]] && [[ ${REBOOT_CAUSE} != null ]]; then
+    echo "both options - force reboot and reboot cause - specified"
+    exec /sbin/reboot -f
+else echo "No appropriate match found, exiting"
+fi
```

Comment:
cases can be reduced here, -r specified versus -r -f specified versus default to `/sbin/reboot $@`
/sbin/reboot should be called even if nothing matches as router will not be in operational state at this point, so no use aborting/exiting at this point

Reply:
Taken care in revised PR change set.